FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,62 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case let .success((data, urlResponse)):
+				guard urlResponse.statusCode == 200 else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+
+				self?.decodeAndComplete(data, completion: completion)
+			}
+		}
+	}
+
+	private func decodeAndComplete(_ data: Data, completion: (FeedLoader.Result) -> Void) {
```

Comment:
Your function here does multiple things. The method name containing an `And` gives this away. Try to move the detail of the completion block out of this function and keep it exclusive to the `RemoteFeedLoader`s `load` method. Instead of completing you could return the `FeedLoader.Result` here like this:
```suggestion
	private func decode(_ data: Data, _ response: HTTPURLResponse) -> FeedLoader.Result {
```

Furthermore you could declare this function static, so you don't need a reference to `self` within the closure passed to the client's `get` method inside the `load`method.